compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -90,6 +93,31 @@ MaterializeEncodingTypeConverter::getEncodingInfo(RankedTensorType type) const {
   return layoutAttr.getEncodingInfo(type);
 }
 
+MemRefFlatteningConverter::MemRefFlatteningConverter(mlir::MLIRContext *ctx) {
+  addConversion([](MemRefType memrefType) {
+    auto rank = memrefType.getRank();
+
+    // Only handle multiple-ranked memrefs, or statically shaped, contiguous
```

Comment:
Why only static shapes? That ... kinda defeats the point of all this.

Reply:
Oh this is outdated code. I think I should retire this MemRefFlatteningConverter. Because when change memrefs, we also need to change its uses.

Reply:
removed